src/executor/stack/mod.rs

```diff
@@ -61,6 +61,22 @@ impl<'config> StackSubstateMetadata<'config> {
 			},
 		}
 	}
+
+	pub fn gasometer(&self) -> &Gasometer<'config> {
+		&self.gasometer
+	}
+
+	pub fn gasometer_mut(&mut self) -> &mut Gasometer<'config> {
```

Comment:
Why do we need this? I think for tracing we only need to read but not modify the state, right?

Reply:
Yeah.. the problem is without a hook to the EVM both `call_inner` or `create_inner` are not accessible from the outside. 

Those functions are responsible to create the Runtime and call it's `run` function, so without further changes in the EVM we are forced to overwrite those functions' implementation to be able to step from outside - and both `call_inner` and `create_inner` call `gasometer.record_cost()`. To get the same behaviour in the tracing context as in the original execution, we need mutable access to the gasometer.

One simple solution for this would be to modify the EVM itself, and move `Runtime` creation and `self.execute` call outside the `call_inner` and `create_inner` functions..